crates/rethnet_evm/benches/state.rs

```diff
@@ -45,6 +45,100 @@ fn bench_code_by_hash(c: &mut Criterion) {
+
+    #[derive(Debug, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct AccountState {
+        #[serde(deserialize_with = "optional_u64_from_hex")]
+        nonce: Option<u64>,
+        balance: Option<U256>,
+        storage: HashMap<U256, Option<U256>>,
+        code: Option<Bytes>,
+        storage_cleared: bool,
+    }
+
+    bench_sync_state_method(
+        c,
+        "DatabaseCommit:commit",
+        prep_no_op,
+        |state, _number_of_accounts, _, _| {
+            #[allow(unused_variables)]
+            let mut accounts_to_commit: HashMap<Address, Account> = HashMap::new();
+            let json_accounts: HashMap<Address, AccountState> = serde_json::from_str(
+                &std::fs::read_to_string("benches/accounts_changed_in_mainnet_block_17295357.json")
+                    .unwrap(),
+                /* a fresh set of account updates can be retrieved via, eg:
+                    BLOCK=17295357 \
+                        HARDHAT_EXPERIMENTAL_VM_MODE=ethereumjs \
+                        HARDHAT_RUN_FULL_BLOCK_DUMP_STATE_TO_FILE=../../crates/rethnet_evm/benches/accounts_changed_in_mainnet_block_${BLOCK}.json \
+                        sh -c 'yarn ts-node scripts/test-run-forked-block.ts $ALCHEMY_URL $BLOCK'
+                    note that this should be done from the packages/hardhat-core directory.
+                */
+            )
+            .unwrap();
+            for (address, account_state) in json_accounts.iter() {
+                let mut storage: HashMap<U256, StorageSlot> = HashMap::new();
+                for (location, value) in account_state.storage.clone() {
+                    storage.insert(
+                        location,
+                        StorageSlot {
+                            original_value: U256::ZERO, // TODO: something better?
+                            present_value: value.unwrap_or(U256::ZERO),
+                        },
+                    );
+                }
+                let code = account_state
+                    .code
+                    .clone()
+                    .map(|bytes| Bytecode::new_raw(bytes));
+                accounts_to_commit.insert(
+                    *address,
+                    Account {
+                        info: AccountInfo {
+                            balance: account_state.balance.unwrap(),
+                            nonce: account_state.nonce.unwrap(),
+                            code: code.clone(),
+                            code_hash: code.map_or(KECCAK_EMPTY, |code| code.hash()),
+                        },
+                        storage,
+                        storage_cleared: account_state.storage_cleared,
+                        is_destroyed: false,
+                        is_touched: false,
+                        is_not_existing: false,
+                    },
+                );
+            }
+
+            state.commit(accounts_to_commit);
```

Comment:
I'm a bit confused about the fact that the whole function is being benchmarked. Shouldn't only this line be? Or, at the very least, I think the whole deserializing of the fixture shouldn't be included in the benchmark, and I'm not sure about the insertion part.

Reply:
You're right. I think I was just taking Remco's spec too literally (see the issue this PR fixes). In the other benchmark issues he had used a "Setup" section separate from "Measurement", so I guess I just assumed everything under "Benchmark" was the measurement target, but I should have thought about it more. Addressed in https://github.com/NomicFoundation/hardhat/pull/3953/commits/5f2b419cf561933d6e05af2de66a76a32f0e34df